File: scalechain-api-domain/src/main/scala/io/scalechain/blockchain/api/domain/RpcResult.scala
package io.scalechain.blockchain.api.domain

trait RpcResult
case class StringResult(value : String) extends RpcResult
case class NumberResult(value : scala.math.BigDecimal) extends RpcResult



case class RpcResponse(result : Option[RpcResult], error:Option[RpcError], id:String)

File: scalechain-api/src/main/kotlin/io/scalechain/blockchain/api/domain/RpcResult.kt
package io.scalechain.blockchain.api.domain
import com.google.gson.*

interface RpcResult
// Return JsNull. (for submitblock)
//data class NullResult() : RpcResult
data class StringResult(val value : String) : RpcResult
data class StringListResult(val value : List<String>) : RpcResult
data class NumberResult(val value : java.math.BigDecimal) : RpcResult

data class RpcResponse(val result : RpcResult?, val error : RpcError?, val id : Long)

